### PR TITLE
Add zendframework/zend-hydrator v3 support and PHP 7.3

### PR DESCRIPTION
**Pull request by webimpress**
Add support for `zendframework/zend-hydrator` v3
Updated also matrix configuration to run tests on PHP 7.3

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
         "phpunit/phpunit": "^7.0.1",
         "zendframework/zend-coding-standard": "~1.0.0",
         "zendframework/zend-expressive-helpers": "^5.0.0alpha3",
-        "zendframework/zend-hydrator": "^2.3.1",
+        "zendframework/zend-hydrator": "^2.3.1 || ^3.0",
```
Comment (Ocramius) [inline composer.json]: Would it make sense to just bump this to `^3.0` instead? I don't see a reason for supporting both releases.

File: test/Assertions.php [hunk 2]
```diff
@@ -21,6 +24,13 @@
 
 trait Assertions
 {
+    public static function getObjectPropertyHydratorClass() : string
+    {
+        return class_exists(ObjectPropertyHydrator::class)
```
Comment (Ocramius) [inline test/Assertions.php]: This can be dropped if we switch to `^3.0` only.

**Comment by weierophinney** [comment]
> As noted in the comments, I wouldn't support both versions of `zendframework/zend-hydrator`: a bump is OK IMO

I don't think we should do that. zend-hydrator 3 depends on PHP 7.2, while this package currently also supports PHP 7.1. Bumping the PHP version here doesn't make sense, IMO.

**Comment by Ocramius** [comment]
@weierophinney I'll agree to disagree (**AGAIN**) then - bumping is fine and encouraged anyway.

PHP 7.1 is in security-only, and 7.3 is out meanwhile: https://secure.php.net/supported-versions.php

I realise that this may be done for stability yadda yadda, but I disagree with doing that unless someone is paying big bucks to keep EOL'd versions there.

File: composer.json (hunk 1 above)
Comment (webimpress) [reply composer.json]: @Ocramius I think we NEED support BOTH versions as `zend-expressive-hal` supports PHP 7.1 and `zend-hydrator` v3 supports only PHP 7.2.

File: test/Assertions.php (hunk 2 above)
Comment (webimpress) [reply test/Assertions.php]: See comment above :)

File: composer.json (hunk 1 above)
Comment (Ocramius) [reply composer.json]: See https://github.com/zendframework/zend-expressive-hal/pull/51#issuecomment-446246085 - I'm in the "drop 7.1 camp". This just means that approval/merge goes through somebody else though ;-)

Comment (webimpress) [reply composer.json]: Ok, missed the discussion below :) Thanks 👍

**Pull request merged**